vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/HttpRequestImpl.java

```diff
@@ -145,6 +147,18 @@ public MultiMap headers() {
     return headers;
   }
 
+  @Override
+  public HttpRequest<T> basicAuthentication(String id, String password) {
```

Comment:
overload this method with an `enc` to specify an encoding to use.

Instead of using a byte array you can use `Buffer.buffer(...)` which has can do the encoding lookup for you

Reply:
I've used `Buffer` in the last commit.

Regarding the overloading with `enc`, it seems that in Java `String` are always using `UTF-16` encoding, do you see a specific case were specifying `enc` is resolving a need?

Reply:
enc is only used for converting the string to a byte[] using a specific encoding. Perhaps we should just overload it with Buffer ? i.e basicAuthentication(Buffer id, Buffer password)